From f8b5688caf50edd2b6a07948032ec30f6b3e2b7c Mon Sep 17 00:00:00 2001
Message-ID: <f8b5688caf50edd2b6a07948032ec30f6b3e2b7c.1763133105.git.jdenemar@redhat.com>
From: Michal Privoznik <mprivozn@redhat.com>
Date: Mon, 29 Sep 2025 13:50:18 +0200
Subject: [PATCH] qemu_command: Prefer virBufferAddLit() in
 qemuBuildCpuHypervCommandLine()
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Using virBufferAsprintf() just to concatenate two literal strings
is excessive. Use virBufferAddLit().

Signed-off-by: Michal Privoznik <mprivozn@redhat.com>
Reviewed-by: Ján Tomko <jtomko@redhat.com>
(cherry picked from commit c66bbac8d56df24674ab15202e4b5611766524d7)
Resolves: https://issues.redhat.com/browse/RHEL-122930
Signed-off-by: Michal Privoznik <mprivozn@redhat.com>
---
 src/qemu/qemu_command.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/qemu/qemu_command.c b/src/qemu/qemu_command.c
index 705d5ab1a1..13e23c7ee3 100644
--- a/src/qemu/qemu_command.c
+++ b/src/qemu/qemu_command.c
@@ -6292,7 +6292,7 @@ qemuBuildCpuHypervCommandLine(virBuffer *buf,
         break;
 
     case VIR_DOMAIN_HYPERV_MODE_PASSTHROUGH:
-        virBufferAsprintf(buf, ",hv-%s=on", "passthrough");
+        virBufferAddLit(buf, ",hv-passthrough=on");
         break;
 
     case VIR_DOMAIN_HYPERV_MODE_NONE:
-- 
2.51.1
